src/app/Component/Common/Options.js

```diff
@@ -1,14 +1,29 @@
-import React, { Component } from 'react';
+import React,{ Component } from 'react';
+import { connect } from 'react-redux';
 import './Header.css';
+import { Link } from 'react-router-dom';
+import { logOut } from '../../actions/actions';
 
-export default class Options extends Component {
+class Options extends Component {
+    constructor(props){
+        super(props);
+    }
+    logOut=()=>{
+        this.props.logOut();  
+    }
     render() {
         return (
             <div className="options">
                 <div className="option-item" onClick={this.props.showProfile}>Profile</div>
                 <div className="option-item">Add to archieve</div>
-                <div className="option-item-logout">Logout</div>
+                <Link className="option-item-logout" to='/' onClick={this.logOut}>Logout</Link>
```

Comment:
navigation should be called after successful logout.

Reply:
Since the options component is child componenet it was giving undefined for history.push so i used link tag 

now i got to know by using withRouter we make availabale of history in chiled componenet now im implementing in this way.